### PR TITLE
Ingestion fixes, auto-create vulnerabilities using upsert

### PR DESCRIPTION
**Pull request by ctron**
This PR adds a few tests for re-ingesting documents. It also fixes issues alongside those. Most notably, it auto-creates vulnerabilities and switches the creation of those to an upsert model.

### CODE REVIEW

File: modules/ingestor/src/graph/organization.rs [hunk 1]
```diff
@@ -40,10 +41,24 @@ impl<'g> OrganizationContext<'g> {
     }
 }
 
-impl super::Graph {
+impl Graph {
+    #[instrument(skip(self, tx), err)]
+    pub async fn get_organizations(
```
Comment (bobmcwhirter) [inline modules/ingestor/src/graph/organization.rs:46]: related: I'd still argue we should be deleting get_plurals() and defer that ish to the appropriate fundamental Service.

**Review by dejanb — APPROVED** [review]

**Review by bobmcwhirter — APPROVED** [review]
We also need some changes in the fundamental VulnerabilityService to avoid the stupid-ass workaround I put there for synthesizing vulnerabilities from the advisory_vuln table when no row existed in the vuln table.

**Comment by ctron** [comment]
@bobmcwhirter I updated the PR and dropped the special handling for the advisory_vulns.

File: modules/ingestor/src/graph/organization.rs (hunk 1 above)
Comment (ctron) [reply modules/ingestor/src/graph/organization.rs:46]: Created a tracking issue: https://github.com/trustification/trustify/issues/547

**Pull request merged**